clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -808,7 +808,8 @@ public void subscribe(Collection<String> topics, ConsumerRebalanceListener liste
                 }
                 log.debug("Subscribed to topic(s): {}", Utils.join(topics, ", "));
                 this.subscriptions.subscribe(topics, listener);
-                metadata.setTopics(subscriptions.groupSubscription());
+                this.metadata.needMetadataForAllTopics(true);
```

Comment:
@hachikuji Thanks. I'm trying to consume, and update the code as per, your suggestion above. I'll reach out to you on Google if I need help.